02_bubble/alfredxiao/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func makeCopy(ns []int) []int {
+	return append([]int{}, ns...)
+}
+
+func bubble(ns []int) []int {
+	copy := makeCopy(ns)
+	n := len(copy)
+	for i := n - 1; i > 0; i-- {
+		for j := 0; j < i; j++ {
+			maybeSwap(copy, j, j+1)
```

Comment:
I agree with @n0npax - awkward name, does more than you want here (ignored returned value).
IMO it'd be more readable if you just inlined :
```go
if copy[j] > copy[j+1] {
    copy[j], copy[j+1] = copy[j+1], copy[j]
}
```
Also: `copy` is a builtin identifier that you are shadowing here.